proxy/dialer_http_connect.go

```diff
@@ -41,28 +41,27 @@ type dialerHTTPConnect struct {
 	forwardAddress string
 }
 
-// Dial makes actual connection to specified address through intermediate HTTP proxy
-func (dialer *dialerHTTPConnect) Dial(network, address string) (conn net.Conn, err error) {
-	conn, err = dialer.forwardDialer.Dial(network, dialer.forwardAddress)
-	if err != nil {
-		return nil, err
-	}
-
-	err = dialer.connectTo(conn, address)
-	if err != nil {
-		conn.Close()
-		return nil, err
-	}
+type Connection struct {
+	net.Conn
+}
 
-	return conn, nil
+// Dial makes actual connection to specified address through intermediate HTTP proxy
+func (dialer *dialerHTTPConnect) Dial(network, address string) (net.Conn, error) {
+	conn, err := dialer.forwardDialer.Dial(network, dialer.forwardAddress)
+	return &Connection{conn}, err
 }
 
-func (dialer *dialerHTTPConnect) connectTo(conn net.Conn, address string) error {
+func (c *Connection) ConnectTo(conn net.Conn, address string, userID string) error {
 	req := &http.Request{
 		Method: "CONNECT",
 		URL:    &url.URL{Host: address},
 		Host:   address,
 	}
+	if len(userID) > 0 {
```

Comment:
Sending `UserID` always will make traffic stick to the single provider only. 
I want to keep an option for round-robin balancing when stickiness is not required.

Reply:
Do we really want to hit round-robin flow? At this stage I was thinking we want to see if one provider gets too much traffic, so that we solve those problems

Reply:
I see round-robin as a default behavior and stickiness as an extra option.